shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPlugin.mm

```diff
@@ -0,0 +1,134 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPlugin.h"
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIApplication.h>
+#import <UIKit/UIKit.h>
+
+#import "flutter/fml/logging.h"
+#import "flutter/shell/platform/darwin/ios/framework/Source/FlutterViewController_Internal.h"
+
+// Method Channel name to start spell check.
+static NSString* const kInitiateSpellCheck = @"SpellCheck.initiateSpellCheck";
+
+@interface FlutterSpellCheckPlugin ()
+
+@property(nonatomic, assign) FlutterMethodChannel* methodChannel;
+@property(nonatomic, retain) UITextChecker* textChecker;
+
+@end
+
+@implementation FlutterSpellCheckPlugin
+
+- (instancetype)initWithMethodChannel:(FlutterMethodChannel*)methodChannel {
+  self = [super init];
+  if (self) {
+    [_methodChannel setMethodCallHandler:^(FlutterMethodCall* call, FlutterResult result) {
+      [self handleMethodCall:call result:result];
```

Comment:
I guess this isn't a retain cycle in MRC since it doesn't increase the retain count. Does it need to check if `self` is nil? Brain hurts.

Reply:
Good catch! We should check if self is nil.